controllers/ibmpowervscluster_controller_test.go

```diff
@@ -839,6 +869,454 @@ func TestIBMPowerVSClusterReconciler_delete(t *testing.T) {
+		g := NewWithT(t)
+		clusterScope = powervsClusterScope()
+		clusterScope.IBMPowerVSCluster.Status = infrav1beta2.IBMPowerVSClusterStatus{
+			ServiceInstance: &infrav1beta2.ResourceReference{
+				ID: ptr.To("serviceInstanceID"),
+			},
+			LoadBalancers: map[string]infrav1beta2.VPCLoadBalancerStatus{
+				"lb": {
+					ID:                ptr.To("lb-id"),
+					ControllerCreated: ptr.To(true),
+				},
+			},
+		}
+		mockPowerVS = powervsmock.NewMockPowerVS(gomock.NewController(t))
+		mockPowerVS.EXPECT().WithClients(gomock.Any())
+		clusterScope.IBMPowerVSClient = mockPowerVS
+		mockResourceClient = resourceclientmock.NewMockResourceController(gomock.NewController(t))
+		clusterScope.ResourceClient = mockResourceClient
+		mockTransitGateway = tgmock.NewMockTransitGateway(gomock.NewController(t))
+		clusterScope.TransitGatewayClient = mockTransitGateway
+		mockVpc = vpcmock.NewMockVpc(gomock.NewController(t))
+		mockVpc.EXPECT().GetLoadBalancer(gomock.Any()).Return(&vpcv1.LoadBalancer{
+			ID:                 ptr.To("lb-id"),
+			Name:               ptr.To("lb"),
+			ProvisioningStatus: ptr.To(string(infrav1beta2.VPCLoadBalancerStateActive)),
+		}, nil, nil)
+		mockVpc.EXPECT().DeleteLoadBalancer(gomock.Any()).Return(&core.DetailedResponse{}, errors.New("failed to delete load balancer"))
+		clusterScope.IBMVPCClient = mockVpc
+		result, err := reconciler.reconcileDelete(ctx, clusterScope)
+		g.Expect(err).To(Not(BeNil()))
+		g.Expect(result.RequeueAfter).To(BeZero())
+	})
+
+	t.Run("when delete LoadBalancer returns requeue as true", func(t *testing.T) {
+		g := NewWithT(t)
+		clusterScope = powervsClusterScope()
+		clusterScope.IBMPowerVSCluster.Status = infrav1beta2.IBMPowerVSClusterStatus{
+			ServiceInstance: &infrav1beta2.ResourceReference{
+				ID: ptr.To("serviceInstanceID"),
+			},
+			LoadBalancers: map[string]infrav1beta2.VPCLoadBalancerStatus{
+				"lb": {
+					ID:                ptr.To("lb-id"),
+					ControllerCreated: ptr.To(true),
+				},
+			},
+		}
+		mockPowerVS = powervsmock.NewMockPowerVS(gomock.NewController(t))
+		mockPowerVS.EXPECT().WithClients(gomock.Any())
+		clusterScope.IBMPowerVSClient = mockPowerVS
+		mockResourceClient = resourceclientmock.NewMockResourceController(gomock.NewController(t))
+		clusterScope.ResourceClient = mockResourceClient
+		clusterScope.TransitGatewayClient = mockTransitGateway
+		mockVpc = vpcmock.NewMockVpc(gomock.NewController(t))
+		mockVpc = vpcmock.NewMockVpc(gomock.NewController(t))
+		mockVpc.EXPECT().GetLoadBalancer(gomock.Any()).Return(&vpcv1.LoadBalancer{
+			ID:                 ptr.To("lb-id"),
+			Name:               ptr.To("lb"),
+			ProvisioningStatus: ptr.To(string(infrav1beta2.VPCLoadBalancerStateDeletePending)),
+		}, nil, nil)
+		clusterScope.IBMVPCClient = mockVpc
+		result, err := reconciler.reconcileDelete(ctx, clusterScope)
+		g.Expect(err).To(BeNil())
+		expectedResult := ctrl.Result{RequeueAfter: time.Minute}
+		g.Expect(result).To(Equal(expectedResult))
```

Comment:
```suggestion
		g.Expect(result).To(Equal(ctrl.Result{RequeueAfter: time.Minute}))
```
How about simplifying it wherever applicable?

Reply:
Done.